tests/tests_all_purge.yml

```diff
@@ -22,7 +22,9 @@
         name: sar-user
 
     - name: Add some mapping
-      shell: echo -e -n "{{ semanage_change }}" | /usr/sbin/semanage -i -
+      shell: |-
+        set -euo pipefail
```

Comment:
Is it possible to `echo` to fail at this point? Or is a general good practice the code needs to follow?

Reply:
> Is it possible to `echo` to fail at this point? Or is a general good practice the code needs to follow?

It isn't possible for `echo` to fail here unless there is some real garbage data in `semanage_change` - but it is possible for `semanage` to fail.

And yes - part of the "penalty" for using the unrecommended, but sometimes necessary, `shell` module, (in fact ansible-lint complains about this and we have to ignore it - `- '305'  # Use shell only when shell functionality is required`) is that we have to use extra checking for the shell commands - so they have a lint rule for forcing the use of `set -o pipefail` when using `shell` - if we're going to do that, we should add `-e` so that shells exit immediately upon error, and `-u` to catch unset variable usage.  Neither one of those cases apply to the above, but we should be consistent and add both of those everywhere.

Reply:
I see, thanks. Would it be possible to update commit messages with these explanations?

Reply:
> I see, thanks. Would it be possible to update commit messages with these explanations?

It depends.  How detailed would you like these explanations to be?